src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -1573,9 +1573,6 @@ public boolean hasFinalField(final DetailAST instanceMember) {
                         result = true;
                     }
                 }
-                if (result) {
```

Comment:
This just ends the loop early when a match is found. So we need to see if we can find a way to have 2 or more matches to cause a difference.

The issue is, all we return from this method if there is a match or not. It doesn't matter if there are multiple, as we will always return the same result. This is just a speed-up process and there is no way to kill this.